README.md

```diff
@@ -278,29 +287,30 @@ A server receiving a client session subscription **MUST** reply back with
+- If session resuming is supported the server will retrieve working values from cache and `result` will have the same id requested by the client. This means a session is "resumed": as a consequence the server **CAN** start pushing jobs omitting to precede them with `mining.set` (see below) and the client **MUST** continue to use values last received within the same session scope. In addition the client **CAN** omit to re-authorize all its workers.
+- If session resuming is supported but the requested session cannot be found for any reason (expiration, cache invalidation), `result` will hold a new session Id which **MUST** be a different value from the `session` member issued by client in previous `mining.subscribe` method. As a consequence the server **MUST** wait for the client to request authorization for its workers and **MUST** send `mining.set` values before pushing jobs. The client **MUST** prepare for a new session discarding all previously cached values (if any).
 
-A server implementing session-resuming **MUST** cache :
-- The session Ids
-- Any active job per session
-- The extraNonce
+A server implementing session-resuming **MUST** cache:
+- Session Ids
 - Any authorized worker
+- Any active job per session
+- ExtraNonce
 
-Servers **MAY** drop entries from the cache on their own schedule. It's up to server to enforce session validation for same agent and/or ip.
+Servers **MAY** drop entries from the cache on their own schedule. It's up to the server to enforce session validation for same agent and/or ip.
 
-A client which successfully subscribes and resumes session (the `session` value in server response is identical to `session` value requested by client on `mining.subscribe`) **CAN** omit to issue the authorization request for it's workers.
+A client which successfully subscribes and resumes session (the `session` value in server response is identical to `session` value requested by client on `mining.subscribe`) **CAN** omit to issue the authorization request for its workers.
 
 ### Session Handling - Noop
-There are cases when a miner struggles to find a solution in a reasonable time so it may trigger the timeout imposed by the server in case of no communications (the server, in fact, may think the client got disconnected). To mitigate the problem a new method `mining.noop`(with no additional parameters) may be requested by the client. 
+There are cases when a miner struggles to find a solution in a reasonable time so it may trigger the timeout imposed by the server in case of no communications (the server may think the client got disconnected). To mitigate the problem a new method `mining.noop`(with no additional parameters) may be sent by the client. 
 ```json
 {
-  "id": 50,
+  "id": 1,
```

Comment:
Let's keep ids random and mathcing with responses.
It seems like requests always have id 1

Reply:
It just feels like this is brittle. If someone comes along later and modifies the spec and does not understand or pay attention to your nomenclature, they could accidentally copy/paste the wrong ID.